hrjobcontract/CRM/Hrjobcontract/BAO/Query.php

```diff
@@ -314,13 +314,21 @@ function whereClauseSingle(&$values, &$query) {
 
   function from($name, $mode, $side) {
     $from = '';
-    
+    $todayDate = date('Y-m-d');
     switch ($name) {
       case 'civicrm_contact':
-        $from .= " $side JOIN civicrm_hrjobcontract hrjobcontract ON hrjobcontract.contact_id = contact_a.id AND hrjobcontract.deleted = 0
-            $side JOIN civicrm_hrjobcontract_revision rev ON rev.jobcontract_id = hrjobcontract.id "
-              . " AND rev.id = (SELECT id FROM civicrm_hrjobcontract_revision WHERE jobcontract_id = hrjobcontract.id "
-              . " AND effective_date <= '" . date('Y-m-d') . "' ORDER BY id DESC LIMIT 1) ";
+        $from .= " $side JOIN ( "
+              . " SELECT hjc.* FROM civicrm_hrjobcontract hjc "
+              . " LEFT JOIN civicrm_hrjobcontract_revision hjr ON hjr.jobcontract_id = hjc.id AND hjr.deleted = 0 "
+              . " LEFT JOIN civicrm_hrjobcontract_details hjd ON hjd.jobcontract_revision_id = hjr.details_revision_id "
+              . " WHERE hjd.period_end_date >= '{$todayDate}' OR hjd.period_end_date IS NULL "
+              . " ORDER BY hjd.period_start_date ASC, hjr.effective_date ASC "
+              . " ) AS hrjobcontract ON hrjobcontract.contact_id = contact_a.id AND hrjobcontract.deleted = 0 "
+              . " $side JOIN civicrm_hrjobcontract_revision rev ON rev.jobcontract_id = hrjobcontract.id "
+              . " AND ("
+              . "   rev.id = (SELECT id FROM civicrm_hrjobcontract_revision WHERE jobcontract_id = hrjobcontract.id "
+              . "   AND (effective_end_date >= '{$todayDate}' OR effective_end_date IS NULL) AND deleted = 0 AND overrided = 0 ORDER BY effective_date ASC LIMIT 1) "
```

Comment:
ORDER BY effective_date ASC is needed to get a first revision which meets the other requirements. Whithout that we could end up getting a future revision even if there is a currently valid revision present.